Makefile

```diff
@@ -0,0 +1,52 @@
+# NOTE: This is only designed for ubuntu and darwin (mac) for now.
+#  I think we could use this if we wanted to expand:
+#    https://stackoverflow.com/questions/714100/os-detecting-makefile
+
+CONDA_ENV_NAME=pickapp
+# Note that the extra activate is needed to ensure that the activate floats env to the front of PATH
+CONDA_ACTIVATE=source $$(conda info --base)/etc/profile.d/conda.sh ; conda activate ; conda activate
+
+ifeq ($(shell uname -s),Darwin)
+	CONDA_YML="environment_mac.yml"
+else ifeq ($shell uname -s,ubuntu)
+	CONDA_YML="environment_ubuntu.yml"
+else
+	@echo "Error, only designed for Darwin & ubuntu currently"
+	exit 0
+endif
+
+# Check for existing conda env.
+ifeq (,$(shell which conda))
+    HAS_CONDA=False
+else
+    HAS_CONDA=True
+    ENV_DIR=$(shell conda info --base)
+    MY_ENV_DIR=$(ENV_DIR)/envs/$(CONDA_ENV_NAME)/
+endif
+
+
+# This dummy timestamp file will allow us to only refresh the environemnt if and when
+# either of the environments are touched.
+data/.create_env: environment_ubuntu.yml environment_mac.yml
```

Comment:
FYI: what make will do is check if either of the files that are listed after the colon have been updates since the last time this step was run. If they were it will know to update this step. This is how I am forcing the environment to be updated any time either of these yaml files are changed.